beacon-chain/core/light-client/lightclient.go

```diff
@@ -372,15 +388,18 @@ func BlockToLightClientHeader(
 		var payloadProof [][]byte
 
 		if blockEpoch < params.BeaconConfig().CapellaForkEpoch {
-			var ok bool
-
-			p, err := execution.EmptyExecutionPayload(version.Capella)
-			if err != nil {
-				return nil, errors.Wrap(err, "could not get payload header")
-			}
-			payloadHeader, ok = p.(*enginev1.ExecutionPayloadHeaderCapella)
-			if !ok {
-				return nil, errors.Wrapf(err, "payload header type %T is not %T", payloadHeader, &enginev1.ExecutionPayloadHeaderCapella{})
+			payloadHeader = &enginev1.ExecutionPayloadHeaderCapella{
+				ParentHash:       make([]byte, fieldparams.RootLength),
+				FeeRecipient:     make([]byte, fieldparams.FeeRecipientLength),
+				StateRoot:        make([]byte, fieldparams.RootLength),
+				ReceiptsRoot:     make([]byte, fieldparams.RootLength),
+				LogsBloom:        make([]byte, fieldparams.LogsBloomLength),
+				PrevRandao:       make([]byte, fieldparams.RootLength),
+				ExtraData:        make([]byte, 0),
+				BaseFeePerGas:    make([]byte, fieldparams.RootLength),
+				BlockHash:        make([]byte, fieldparams.RootLength),
+				TransactionsRoot: make([]byte, fieldparams.RootLength),
+				WithdrawalsRoot:  make([]byte, fieldparams.RootLength),
 			}
```

Comment:
The `execution.EmptyExecutionPayload(version.Capella)` returns something that is different than `enginev1.ExecutionPayloadHeaderCapella` and it would throw an error when casting. so I just created one here. and it's only for this case.

Reply:
> The `execution.EmptyExecutionPayload(version.Capella)` returns something that is different than `enginev1.ExecutionPayloadHeaderCapella` and it would throw an error when casting. so I just created one here. and it's only for this case.

The `execution.EmptyExecutionPayload()` function returns an empty payload, not the payload header. There's no function for the latter, but ig we can implement it.

Reply:
from what I saw in the codebase now, the `execution.EmptyExecutionPayload()` can also be used to generate empty payload headers. I think after the typecast, the `txsRoot` and `wxRoots` are set to empty by default. In that case we can just use the existing function.